### PR TITLE
chore: Navigation analytics

### LINKED ISSUE #18299: [Task]: cmd+click analytics events
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- [x] trigger an event when a user uses cmd+click
- [x] trigger an event when a user clicks on an action on the Entity explorer. (this will be used to figure out how users are navigating)
- [x] If not existing, trigger an event from the omnibar as well, for when a user navigates to an action

### LINKED ISSUE #18810: [Bug]: Recent Entities is not accurate
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The recent entities stored and shown on omni bar are not correct. They seem to skip and few and add ones not visited at all

### Steps To Reproduce

1. Navigate to certain actions/widgets/js objects
2. Check omnibar top 5 list
3. They do not match the ones recently visited

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Prod

### PR DESCRIPTION
**Pull request by hetunandu**
## Description

Adds extra information to the ROUTE_CHANGE analytics event
- Adds where the navigation comes from
- Adds what kind of entity is being navigated to
- Adds if it is one of the recent entities
- Adds which ui element the navigation was invoked from eg Entity Explorer or Debugger

Also fixes recent entities not being accurate

Fixes #18299
Fixes #18810 

## Type of change

- New feature (non-breaking change which adds functionality)

- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/AppRouter.tsx [hunk 1]
```diff
@@ -101,15 +100,12 @@ function AppRouter(props: {
...
-    const stopListener = history.listen((location: any) => {
-      AnalyticsUtil.logEvent("ROUTE_CHANGE", { path: location.pathname });
-      changeAppBackground(props.currentTheme);
-    });
     getCurrentUser();
     getFeatureFlags();
     getCurrentTenant();
-    return stopListener;
+    return () => {
+      changeAppBackground(props.currentTheme);
```
Comment (akash-codemonk) [inline app/client/src/AppRouter.tsx:107]: This should be called on route change no?

Comment (hetunandu) [reply app/client/src/AppRouter.tsx:107]: It used to be called when unmounted. That is still happening

Comment (akash-codemonk) [reply app/client/src/AppRouter.tsx:107]: Hey no I think `history.listen` returns a call back(`stopListener` here) for unsubscribing from the listener which used to get called on unmount.

Comment (hetunandu) [reply app/client/src/AppRouter.tsx]: I have updated the code to do this on route change

**Review by akash-codemonk — DISMISSED** [review]

**Review by akash-codemonk — APPROVED** [review]

**Pull request merged**